env/env_test.cc

```diff
@@ -579,142 +593,127 @@ TEST_P(EnvPosixTestWithParam, DecreaseNumBgThreads) {
                    Env::Priority::HIGH);
     Env::Default()->SleepForMicroseconds(kDelayMicros);
   }
-  ASSERT_EQ(2U, env_->GetThreadPoolQueueLen(Env::Priority::HIGH));
-  ASSERT_TRUE(tasks[0].IsSleeping());
-  ASSERT_TRUE(!tasks[1].IsSleeping());
-  ASSERT_TRUE(!tasks[2].IsSleeping());
+  SleepUntilQueueLen(Env::Priority::HIGH, 2U);
+  ASSERT_TRUE(tasks[0].ShortWaitForSleeping());
+  ASSERT_TRUE(tasks[1].ShortWaitForWokenUp());
+  ASSERT_TRUE(tasks[2].ShortWaitForWokenUp());
 
   // Increase to 2 threads. Task 0, 1 running; 2 waiting
   env_->SetBackgroundThreads(2, Env::Priority::HIGH);
-  Env::Default()->SleepForMicroseconds(kDelayMicros);
-  ASSERT_EQ(1U, env_->GetThreadPoolQueueLen(Env::Priority::HIGH));
-  ASSERT_TRUE(tasks[0].IsSleeping());
-  ASSERT_TRUE(tasks[1].IsSleeping());
-  ASSERT_TRUE(!tasks[2].IsSleeping());
+  SleepUntilQueueLen(Env::Priority::HIGH, 1U);
+  ASSERT_TRUE(tasks[0].ShortWaitForSleeping());
+  ASSERT_TRUE(tasks[1].ShortWaitForSleeping());
+  ASSERT_TRUE(tasks[2].ShortWaitForWokenUp());
 
   // Shrink back to 1 thread. Still task 0, 1 running, 2 waiting
   env_->SetBackgroundThreads(1, Env::Priority::HIGH);
-  Env::Default()->SleepForMicroseconds(kDelayMicros);
-  ASSERT_EQ(1U, env_->GetThreadPoolQueueLen(Env::Priority::HIGH));
-  ASSERT_TRUE(tasks[0].IsSleeping());
-  ASSERT_TRUE(tasks[1].IsSleeping());
-  ASSERT_TRUE(!tasks[2].IsSleeping());
+  SleepUntilQueueLen(Env::Priority::HIGH, 1U);
```

Comment:
I don't think we need to wait here. We can directly assert the length of the queue.